client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/hooks/use-recipe.ts

```diff
@@ -38,105 +66,127 @@ const useRecipe = ( siteId = 0, patterns: Pattern[], categories: Category[] ) =>
 		enabled: !! font_variation_title,
 	} );
 
-	const { setSelectedDesign } = useDispatch( ONBOARD_STORE );
-
-	const selectedDesignRef = useRef< Design | undefined >( selectedDesign );
+	const colorVariation =
+		( colorVariations || [] ).find( ( { title } ) => title === color_variation_title ) || null;
+
+	const fontVariation =
+		( fontVariations || [] ).find( ( { title } ) => title === font_variation_title ) || null;
+
+	const setHeader = ( pattern: Pattern | null ) => {
+		setSearchParams(
```

Comment:
I guess setting the params when selecting any patterns might cause some redundant re-rendering 🤔

Reply:
Can you clarify what this code breaks? React re-renders are normal thing; it doesn't have to update the DOM. The pattern interaction / animation is still smooth because we preserve the pattern keys in this PR.